### DISCUSSION TITLE
Two questions and request for guidance regarding observation weighting and prior covariance matrices

### DISCUSSION THREAD

**Discussion by d-strom** [gmdsi/GMDSI_notebooks#121] [General]
Hi,

Let me first apologize in advance for the long wall of text. (TLDR; 1. how does one handle weighting of a single observation group or when there is a lack of relevant observation data. 2. What is the purpose of prior covariance matrices or noptmax -1?)

I've been looking through some of the recently released YouTube material on the GMDSI notebooks (which are great by the way). Something that is empathized a lot in both the written content as well as the video-tutorials is the importance of weighting.
The process of weighting overall seems to focus a whole lot on observation groups rather than individual observations. That is making the weights balanced on a group by group basis such that flows and head observations play an equal role in the process of minimizing the objective function.
This, of course, assumes that there are multiple observation groups. Something that I feel isn't necessarily a fact. This leads me to two questions.

1. How do I apply weights in a relevant way with something like the "balanced groups" approach for a single observation group comprised of many observations and observation locations? is it even possible? let alone a good way to do it?
2. If you can't. What are the best practices to apply weighting for few or a single observation group? Is it even necessary?

Naturally, this prompts the following question: Why do you only have one observation group?
Well, I guess multiple observation groups is highly preferable in almost all cases. But I find this to be a rare thing in the real world to achieve. Many times one could have both point head observations and stream flows etc. But quite often when working in an urban environment you have - at best - head observations and possibly a few measurements of flow/seepage into some known construction. In the model domain there might not be any streams or wells with extraction. The recharge is probably known to some degree, say within a factor 0.25 or 0.5 around some average value on an annual basis.

How do others deal with this type of situation (which in my mind is the normal situation) when it comes to inverse calibration using the pest++ suite?

And to my second question. Something I also feel like is pushed for a lot is the importance of running prior monte carlo. Could someone help me understand the reason for this? What is the difference for instance between running prior monte carlo to construct a covariance matrix, read that into my pest++ control file and run pestpp-ies from there as opposed to just running pestpp-ies right of the bat?
I notice also you could run the pestpp-ies with noptmax set to -1 to build a jacobian matrix and give the model a "head start" in the inverse calibration process.

I guess I'm just a bit confused overall to differentiate between the output of a prior monte carlo, prior covariance matrix and jacobian matrix. As I understand, pestpp-ies builds all these files on its own at the start of the first iteration anyways. Naturally there must be some distinction between the two approaches but I feel like it isn't entirely clear to me. If anyone could explain this (again) in as simple terms as possible I would greatly appreciate it.

Sincerely
David

**Comment by jtwhite79** [comment]
I think there might need to be clarity around what groups are and how they are used.  Groups can be any arbitrary grouping of observations you want, not necessarily different data types.  I commonly put all obs from a specific site into a group or even break those down into different time periods (for example, wet-season and dry-season observations at the same site could "site1-summer" and "site1-winter", etc).  Or you could split into geographic regions like "urban" and "rural" gw levels, etc.   The result of all this grouping is so that you can monitor the performance/response of each group, and you can more effectively balance the contribution of each of the important (to you) groupings to the composite objective function to bring out more of the important signal components in the observation dataset.  We should probably demonstrate some of this grouping in the notebooks...

The reason to run prior monte carlo is to see how well the model performs across a range expected inputs before trying to do parameter adjustment.  If you run model one time with best initial parameter values, you dont get to see how much output variability can be induced by expected parameter uncertainty, and you dont get to see how the obs vs sim stacks up in the context of this expected uncertainty.  

For example, running the model once, you could see this (grey is sim, red is obs)

![image](https://github.com/gmdsi/GMDSI_notebooks/assets/1770912/427e467a-c6ce-4235-966c-044bbb10035e)

Is that an acceptable obs vs sim pattern for an uncalibrated model?  Maybe?  maybe not? Who knows - there is no context.  But with a trivially easy prior monte carlo (say 50 model runs), you can make this plot:

![image](https://github.com/gmdsi/GMDSI_notebooks/assets/1770912/c47ebcef-ffb7-4847-90a2-89aff41fb8e7)

Now its obvious something is off - in late time, the simulate results dont cover the observed values, even with maximum expected parameter uncertainty.  For me, this kind of context has a huge return on investment.

On the cov vs prior monte carlo vs jcb, the cov is matrix that can be stored in either an ASCII ".mat" or ".cov" or ".unc" format or can be stored in binary format, in which case we just repurpose the ".jcb" format to store it (since it usually very sparse).  We can also store the prior parameter ensemble (drawn from the prior cov matrix) is ASCII (".csv") or binary (again we can repurpose the ".jcb" format to store this ensemble).  Prior monte carlo is an operation, the result of which is an observation ensemble that can be stored in ".csv" or ".jcb".  So the ".jcb" is just sparse matrix compressed binary file format.  thats all.

**Reply by d-strom** [reply]
Thanks for the clarifications Jeremy. I agree with the concepts your describing and which you are neatly presenting with your two diagrams (how could I not?!). Seems very reasonable and provides good metrics for evaluation.

I think I also understand what you mean by the response to different observation groups and how this concept can be used to subdivide observations into different sets of relevant subgroups depending on the model.

What I dont quite understand still is the latter part. The prior covariance matrix, is its purpose to smooth out the transition between different pilotpoints, for instance, to fit certain geostatistical patterns? Wile the jacobian matrix/monte carlo parts are ways to inform my model of different sets of parameters before running -ies or other methods to give me some idea of variation and provide pest with means to gain a head start in the calibration process?

Or are they essentially the same thing but expressed with different words depending on context? Perhaps I might be barking up the wrong tree here.. But I'm grateful for the support.
